src/formatDate.js

```diff
@@ -8,7 +8,28 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[3];
+  const partsOfDate = date.split(oldSeparator);
+  const dateObject = {};
+
+  for (let i = 0; i < 3; i++) {
+    dateObject[fromFormat[i]] = partsOfDate[i];
+  }
+
+  const newSeparator = toFormat[3];
+  const newPartsOfDate = toFormat.slice(0, 3).map((part) => {
+    if (part === 'YY' && dateObject['YYYY']) {
+      return dateObject['YYYY'].slice(2);
+    } else if (part === 'YYYY' && dateObject['YY']) {
+      const year = parseInt(dateObject['YY'], 10);
+
+      return year < 30 ? `20${dateObject['YY']}` : `19${dateObject['YY']}`;
```

Comment:
The year conversion logic is correct. However, when dealing with years, it might be more future-proof to consider a more flexible approach that doesn't hardcode the century pivot year (30) or the centuries (20 and 19). This way, if the pivot year changes, only one place in the code needs to be updated.